src/rules/rename-default-import.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview Rule to enforce aliases for default imports
+ * @author Michał Kołodziejski
+ */
+
+import docsUrl from '../docsUrl'
+
+
+function isDefaultImport(specifier) {
+  if (specifier.type === 'ImportDefaultSpecifier') {
+    return true
+  } else if (specifier.type === 'ImportSpecifier' && specifier.imported.name === 'default') {
+    return true
+  } else {
+    return false
+  }
+}
+
+
+function handleImport(context, specifier, source) {
+  const {value: packageName} = source
+  const {local: {name: importAlias}} = specifier
+  const mappings = context.options[0]
+
+  if (Object.keys(mappings).indexOf(packageName) === -1) {
+    return
+  }
+
+  if (mappings[packageName] !== importAlias) {
+    context.report({
+      node: source,
+      message: `Default import from '${packageName}' should be aliased to `
+               + `${mappings[packageName]}, not ${importAlias}`,
+      fix: fixer => {
+        let newAlias = mappings[packageName]
+        if (specifier.imported && specifier.imported.name === 'default') {
+          newAlias = `default as ${mappings[packageName]}`
+        }
+
+        return fixer.replaceText(specifier, newAlias)
+      },
+    })
+
+    const declaredVariable = context.getDeclaredVariables(specifier)[0]
```

Comment:
Why is this “the first” of the declared variables?

Reply:
so, according to `eslint` docs

> If the node is an ImportSpecifier, ImportDefaultSpecifier, or ImportNamespaceSpecifier, the declared variable is returned.

I care about the binding we create here so that later I can get all the references to that binding used in the code.

Reply:
sorry, to clarify - why is this returning an array, but you only care about the first item? is it always an array of length 1?

Reply:
That depends on the type of the argument passed to the method as far as I understood